akka-actor-typed/src/main/scala/akka/actor/typed/javadsl/ActorContext.scala

```diff
@@ -167,4 +171,34 @@ trait ActorContext[T] {
    */
   def spawnAdapter[U](f: JFunction[U, T]): ActorRef[U]
 
+  /**
+   * Perform a single request-response message interaction with another actor, and transform the messages back to
+   * the protocol of this actor.
+   *
+   * The interaction has a timeout (to avoid resource a resource leak). If the timeout hits without any response it
+   * will be passed as an [[akka.pattern.AskTimeoutException]] to the `responseToOwnProtocol` function.
```

Comment:
Perhaps we should use `java.util.concurrent.TimeoutException` instead

Reply:
I agree, good idea.